pkg/agent/controller/egress/egress_controller.go

```diff
@@ -628,9 +648,18 @@ func (c *EgressController) syncEgress(egressName string) error {
 		eState = c.newEgressState(egressName, egress.Spec.EgressIP)
 	}
 
-	localNodeSelected, err := c.cluster.ShouldSelectIP(egress.Spec.EgressIP, egress.Spec.ExternalIPPool)
-	if err != nil {
-		return err
+	localNodeSelected := false
+	if egress.Spec.ExternalIPPool != "" {
+		maxEgressIPsFilter := func(node string) bool {
+			egressesOnNode, _ := c.egressInformer.GetIndexer().IndexKeys(egressNodeIndex, node)
+			egressSet := sets.NewString(egressesOnNode...)
+			egressSet.Insert(egress.Name)
+			return egressSet.Len() <= c.maxEgressIPsPerNode
```

Comment:
Can we infer the actual assigned IPs from the number of distinct egress names by indexing `egressNodeIndex`? IIRC egress also supports assigning Node IPs as Egress IPs without allocating them from the external IP pool.

Reply:
Great catch. Yes, we should only count the number of distinct egress IPs that are allocated from ExternalIPPool.